lib/ast/AST.cpp

```diff
@@ -150,21 +150,32 @@ bool KOREObjectSymbol::isBuiltin() const {
 void KOREObjectSymbol::instantiateSymbol(KOREObjectSymbolDeclaration *decl) {
   std::vector<KOREObjectSort *> instantiated;
   int i = 0;
-  std::unordered_set<KOREObjectSortVariable, HashSort> vars;
+  std::unordered_map<KOREObjectSortVariable, size_t, HashSort> vars;
   for (auto var : decl->getObjectSortVariables()) {
-    vars.emplace(*var);
+    vars.emplace(*var, i++);
   }
   for (auto sort : decl->getSymbol()->getArguments()) {
     if (auto var = dynamic_cast<KOREObjectSortVariable *>(sort)) {
       if (vars.count(*var)) {
-        instantiated.push_back(arguments[i++]);
+        instantiated.push_back(arguments[vars.at(*var)]);
       } else {
         instantiated.push_back(sort);
       }
     } else {
       instantiated.push_back(sort);
     }
   }
+  auto returnSort = decl->getSymbol()->sort;
+  if (auto var = dynamic_cast<KOREObjectSortVariable *>(returnSort)) {
```

Comment:
Doesn't this need to substitute for the sort variables everywhere inside the declared sort?
Consider something like `symbol addMapEntry{K,V}(Map{K,V},K,V):Map{K,V}`. `Map{K,V}` isn't a sort variable, so I think it would be left alone.